samples/client/petstore/java/jersey2/src/test/java/io/swagger/client/api/PetApiTest.java

```diff
@@ -1,155 +1,288 @@
 
-        // TODO: test validations
+        api.updatePet(pet);
+
+        List<Pet> pets = api.findPetsByStatus(Arrays.asList(new String[]{"available"}));
+        assertNotNull(pets);
+
+        boolean found = false;
+        for (Pet fetched : pets) {
+            if (fetched.getId().equals(pet.getId())) {
+                found = true;
+                break;
+            }
+        }
+
+        assertTrue(found);
     }
-    
-    /**
-     * Updates a pet in the store with form data
-     *
-     * 
-     *
-     * @throws ApiException
-     *          if the Api call fails
-     */
+
     @Test
-    public void updatePetWithFormTest() throws ApiException {
-        Long petId = null;
-        String name = null;
-        String status = null;
-        // api.updatePetWithForm(petId, name, status);
+    public void testFindPetsByTags() throws Exception {
+        Pet pet = createRandomPet();
+        pet.setName("monster");
+        pet.setStatus(Pet.StatusEnum.AVAILABLE);
+
+        List<Tag> tags = new ArrayList<Tag>();
+        Tag tag1 = new Tag();
+        tag1.setName("friendly");
+        tags.add(tag1);
+        pet.setTags(tags);
+
+        api.updatePet(pet);
+
+        List<Pet> pets = api.findPetsByTags(Arrays.asList(new String[]{"friendly"}));
+        assertNotNull(pets);
 
-        // TODO: test validations
+        boolean found = false;
+        for (Pet fetched : pets) {
+            if (fetched.getId().equals(pet.getId())) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found);
     }
-    
-    /**
-     * uploads an image
-     *
-     * 
-     *
-     * @throws ApiException
-     *          if the Api call fails
-     */
+
+    @Test
+    public void testUpdatePetWithForm() throws Exception {
+        Pet pet = createRandomPet();
+        pet.setName("frank");
+        api.addPet(pet);
+
+        Pet fetched = api.getPetById(pet.getId());
+
+        api.updatePetWithForm(fetched.getId(), "furt", null);
+        Pet updated = api.getPetById(fetched.getId());
+
+        assertEquals(updated.getName(), "furt");
+    }
+
     @Test
-    public void uploadFileTest() throws ApiException {
-        Long petId = null;
-        String additionalMetadata = null;
-        File file = null;
-        // ModelApiResponse response = api.uploadFile(petId, additionalMetadata, file);
+    public void testDeletePet() throws Exception {
+        Pet pet = createRandomPet();
+        api.addPet(pet);
+
+        Pet fetched = api.getPetById(pet.getId());
+        api.deletePet(fetched.getId(), null);
+
+        try {
+            fetched = api.getPetById(fetched.getId());
+            fail("expected an error");
+        } catch (ApiException e) {
+            assertEquals(404, e.getCode());
+        }
+    }
+
+    @Test
+    public void testUploadFile() throws Exception {
+        Pet pet = createRandomPet();
+        api.addPet(pet);
+
+        File file = new File("hello.txt");
```

Comment:
I think we need to keep hello.txt as required by the test above.

But given that the test case passed, it seems to me something is not right as the test is supposed to fail due to hello.txt not present.